src/modules/flow/http-server/http-server.c

```diff
@@ -713,6 +714,167 @@ float_process_cb(struct http_data *mdata, const struct sol_flow_packet *packet)
     return sol_flow_packet_get_drange(packet, &mdata->value.d);
 }
 
+static void
+rgb_send_packet_cb(struct http_data *mdata, struct sol_flow_node *node)
+{
+    sol_flow_send_rgb_packet(node,
+        SOL_FLOW_NODE_TYPE_HTTP_SERVER_RGB__OUT__OUT, &mdata->value.rgb);
+}
+
+static int
+rgb_response_cb(struct http_data *mdata, struct sol_buffer *content, bool json)
+{
```

Comment:
I don't think enforcing the content type is a good idea. As we discussed I will return the color type in hexadecimal if the content type is not json.